### PR TITLE
Json cate

### PR DESCRIPTION
**Pull request by jmafoster1**
Some changes to facilitate the CARLA case study. Most notably, these include rudimentary capability for CATE estimation as well as coefficient estimation, so we don't have to do concrete metamorphic tests.

### CODE REVIEW

File: causal_testing/json_front/json_class.py [hunk 1]
```diff
@@ -188,7 +225,9 @@ def _execute_test_case(self, causal_test_case: CausalTestCase, test: Iterable[Ma
             logger.warning("   FAILED- expected %s, got %s", causal_test_case.expected_causal_effect, result_string)
         return failed
 
-    def _setup_test(self, causal_test_case: CausalTestCase, test: Mapping) -> tuple[CausalTestEngine, Estimator]:
+    def _setup_test(
+        self, causal_test_case: CausalTestCase, test: Mapping, conditions: list[str] = None
```
Comment (cwild-UoS) [inline causal_testing/json_front/json_class.py:229]: Might be nice to add the new `conditions` param to the docstring, but it is optional as this is a private method that users won't be interacting with. 

I believe that pylint does not pick this up as it is private

File: causal_testing/specification/causal_dag.py [hunk 2]
```diff
@@ -521,5 +521,14 @@ def identification(self, base_test_case: BaseTestCase):
         minimal_adjustment_set = min(minimal_adjustment_sets, key=len)
         return minimal_adjustment_set
 
+    def to_dot(self) -> str:
```
Comment (cwild-UoS) [inline causal_testing/specification/causal_dag.py:524]: Should this have a different name? 'to_dot' suggests you may be getting a `.dot` file back not a str?

**Review by cwild-UoS — APPROVED** [review]
Looks really good!

File: causal_testing/json_front/json_class.py (hunk 1 above)
Comment (jmafoster1) [reply causal_testing/json_front/json_class.py:229]: Good spot. Fixed

**Pull request merged**